src/mocks/timeseriesDataPointsList.ts

```diff
@@ -2,8 +2,87 @@ import {
   DatapointsGetAggregateDatapoint,
   DatapointsGetDoubleDatapoint,
 } from '@cognite/sdk';
+import { MockChartDataConfig } from '.';
 
-export function randomData(
+export const defaultConf: MockChartDataConfig = {
+  id: 0,
+  min: 0,
+  max: 150,
+  continousDeviation: 3,
+  peakDeviation: 30,
+};
+
+export function getRandomdata(
+  id: number,
+  start: number,
+  end: number,
+  n: number,
+  granularity?: number,
+  config?: any
+) {
+  let conf = config ? getConig(config, id) : undefined;
+  conf = conf ? conf : defaultConf;
+  const arr = randomData(start, end, n, conf, granularity);
+  return makePeakPoints(arr, conf);
+}
+
+function getConig(config: any[], id: number) {
+  return config.find(conf => conf.id === id);
+}
+
+// tslint:disable-next-line:cognitive-complexity
+function makePeakPoints(
+  resultArray: DatapointsGetAggregateDatapoint,
+  config: MockChartDataConfig
+) {
+  let sortedPoints = [...resultArray.datapoints].sort((dp1, dp2) => {
+    if (dp1.average && dp2.average) {
+      return dp1.average > dp2.average ? -1 : 1;
+    }
+    return -1;
+  });
+
+  sortedPoints = removeDuplicatesBy(
+    (x: { average: any }) => x.average,
+    sortedPoints
+  );
+
+  const { positivePeakPoints, negativePeakPoints, peakDeviation } = config;
+  const positiveDPs = positivePeakPoints ? positivePeakPoints : 0;
+  const negativeDPs = negativePeakPoints ? negativePeakPoints : 0;
+
+  for (let i = 0; i < positiveDPs; i++) {
+    const dp = sortedPoints[i];
+    dp.average = dp.average ? dp.average + peakDeviation : undefined;
+    dp.min = dp.min ? dp.min + peakDeviation : undefined;
+    dp.max = dp.max ? dp.max + peakDeviation : undefined;
+  }
+
+  const dpLength = sortedPoints.length;
+
+  for (let i = 0; i < negativeDPs; i++) {
+    const dp = sortedPoints[dpLength - i - 1];
+    dp.average = dp.average ? dp.average - peakDeviation : undefined;
+    dp.min = dp.min ? dp.min - peakDeviation : undefined;
+    dp.max = dp.max ? dp.max - peakDeviation : undefined;
+  }
+
+  return resultArray;
+}
+
+function removeDuplicatesBy(keyFn: any, array: any) {
```

Comment:
Didn't get the purpose of this function

Reply:
This is used as a helper function for making distinct peak points peak deviation.